### PR TITLE
Add warning to set SENDMAIL_ARGS to -- 

### PR DESCRIPTION
**Pull request by zeripath**
Even with #17688 email addresses that contain an initial `-` may still be present in the db and it may in future still be possible to imagine a situation whereby initial `-` are repermitted.

This PR simply updates the documentation to warn users to set their SENDMAIL_ARGS with a terminal `--` to prevent this possibility email addresses being interpreted as options.

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

File: custom/conf/app.example.ini [hunk 1]
```diff
@@ -1533,7 +1533,7 @@ PATH =
 ;SENDMAIL_PATH = sendmail
 ;;
 ;; Specify any extra sendmail arguments
-;SENDMAIL_ARGS =
+;SENDMAIL_ARGS = ; WARNING: if your sendmail program interprets options you should set this to "--" or terminate these args with "--"
```
Comment (silverwind) [inline custom/conf/app.example.ini]: Would move this to the comment 1 line above and add the actual default `"--"` if it is that.

File: docs/content/doc/usage/email-setup.en-us.md [hunk 2]
```diff
@@ -35,6 +35,7 @@ ENABLED       = true
 FROM          = gitea@mydomain.com
 MAILER_TYPE   = sendmail
 SENDMAIL_PATH = /usr/sbin/sendmail
+SENDMAIL_ARGS="--" ; most "sendmail" programs take options, "--" will prevent an email address being interpreted as an option.
```
Comment (silverwind) [inline docs/content/doc/usage/email-setup.en-us.md]:
```suggestion
SENDMAIL_ARGS = "--"
```
No need to replicate docs here imho.

**Comment by silverwind** [comment]
Maybe we could just automatically add `--` final argument if it's not there in the config?

File: custom/conf/app.example.ini (hunk 1 above)
Comment (zeripath) [reply custom/conf/app.example.ini]: it's not the default.

File: docs/content/doc/usage/email-setup.en-us.md (hunk 2 above)
Comment (zeripath) [reply docs/content/doc/usage/email-setup.en-us.md]:
```suggestion
SENDMAIL_ARGS = "--" ; most "sendmail" programs take options, "--" will prevent an email address being interpreted as an option.
```

Comment (zeripath) [reply docs/content/doc/usage/email-setup.en-us.md]: I don't agree - people don't read docs.

**Comment by zeripath** [comment]
> Maybe we could just automatically add `--` final argument if it's not there in the config?

Are you able to go through and check EVERY "sendmail" that exists and assert that `--` actually terminates options?

If not we would need another option to not append the `--` or perhaps just not put `--` if the SENDMAIL_ARGS is not empty.

We'd also likely need another option which is to refuse to send emails via sendmail if the email address starts with a `-`

**Comment by silverwind** [comment]
> Are you able to go through and check EVERY "sendmail" that exists and assert that -- actually terminates options?

Can't say that for sure but `--` is a very common mechanism for unix programs to terminate arguments. Not sure about stuff like Windows implementations thought. I guess it's fine to not try the route of forcing the parameter.

File: custom/conf/app.example.ini (hunk 1 above)
Comment (silverwind) [reply custom/conf/app.example.ini]: Ok, but comment should still be moved.

**Review by silverwind — APPROVED** [review]

File: docs/content/doc/advanced/config-cheat-sheet.en-us.md [hunk 3]
```diff
@@ -666,7 +666,7 @@ Define allowed algorithms and their minimum key length (use -1 to disable a type
    - Enabling dummy will ignore all settings except `ENABLED`, `SUBJECT_PREFIX` and `FROM`.
 - `SENDMAIL_PATH`: **sendmail**: The location of sendmail on the operating system (can be
    command or full path).
-- `SENDMAIL_ARGS`: **_empty_**: Specify any extra sendmail arguments.
+- `SENDMAIL_ARGS`: **_empty_**: Specify any extra sendmail arguments. (NOTE: you should be aware that email addresses can look like options - if your `sendmail` command takes options you must set the option terminator `--`)
```
Comment (6543) [inline docs/content/doc/advanced/config-cheat-sheet.en-us.md:669]: can we add -- by default? (go code)

**Comment by 6543** [comment]
:rocket: - merge this first ... we can talk about defaults later

**Pull request merged**